src/odemis/util/test/transform_test.py

```diff
@@ -334,10 +334,23 @@ class PixelIndexCoordinateTransformBase(unittest.TestCase):
 
     def setUp(self):
         self._ji = [(0, 0), (0, 4), (7, 0), (7, 4), (0.5, 0.5)]
-        self._xy = [(-2, 3.5), (2, 3.5), (-2, -3.5), (2, -3.5), (-1.5, 3)]
-        self._xy2 = [(-4, 7), (4, 7), (-4, -7), (4, -7), (-3, 6)]
-        self._xy23 = [(-4, 10.5), (4, 10.5), (-4, -10.5), (4, -10.5), (-3, 9)]
-        self._shape = (8, 5)
+
+        # List of known values [(kwargs0, xy0), (kwargs1, xy1), ...] for which
+        # `xy == to_physical_space(self._ji, **kwargs)` is True.
+        self._known_values = [
+            ({'shape': (8, 5)},
+             [(-2, 3.5), (2, 3.5), (-2, -3.5), (2, -3.5), (-1.5, 3)]),
+            ({'shape': (8, 5), 'pixel_size': 2},
+             [(-4, 7), (4, 7), (-4, -7), (4, -7), (-3, 6)]),
+            ({'shape': (8, 5), 'pixel_size': (2, 3)},
+             [(-4, 10.5), (4, 10.5), (-4, -10.5), (4, -10.5), (-3, 9)]),
+            ({},
+             [(0, 0), (4, 0), (0, -7), (4, -7), (0.5, -0.5)]),
+            ({'pixel_size': 2},
+             [(0, 0), (8, 0), (0, -14), (8, -14), (1, -1)]),
+            ({'pixel_size': (2, 3)},
+             [(0, 0), (8, 0), (0, -21), (8, -21), (1, -1.5)]),
+        ]
```

Comment:
This still looks confusing to me. It's especially not clear to me what the list at the end of each entry is. Are those input or output coordinates?

Can you add a short comment why the empty dict on line 347 is necessary? Does that default to `shape=None` and `pixel_size=None`?

Reply:
I agree that this dict can still be improved a bit, for the rest the PR looks good now.

Reply:
improved this bit; hope it's clear now